src/content/docs/en/guides/upgrade-to/v6.mdx

```diff
@@ -840,6 +840,39 @@ If you want to keep the old ID generation for backward compatibility reasons, yo
 
 <ReadMore>Learn more about [Heading IDs](/en/guides/markdown-content/#heading-ids).</ReadMore>
 
+### Changed: numbers are no longer allowed in `params` returned by `getStaticPaths()`.
+
+<SourcePR number="14586" title="fix!: disallow number in getStaticPaths params"/>
+
+Previously, `getStaticPaths()` could return `params` with number values. However, at runtime `Astro.params` values are always strings or undefined because they come from URL segments. This change removes that mismatch so the types reflect runtime behavior.
```

Comment:
```suggestion
In Astro 5.x, `getStaticPaths()` could return `params` of type number, which would always be stringified by Astro. However, that was confusing because it conflicted with `Astro.params` types.

Astro 6.0 removes this behavior: `getStaticPaths()` must now return string or undefined `params` values.
```